lib/active_merchant/billing/gateways/agms.rb

```diff
@@ -0,0 +1,322 @@
+# Agms Gateway uses nokogiri to build the xml request
+require 'nokogiri'
+
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class AgmsGateway < Gateway
+      self.test_url = 'https://gateway.agms.com/roxapi/agms.asmx'
+      self.live_url = 'https://gateway.agms.com/roxapi/agms.asmx'
+
+      self.supported_countries = ['US']
+      self.default_currency = 'USD'
+      self.supported_cardtypes = [:visa, :master, :american_express, :discover, :dinner_club, :jcb]
+
+      self.homepage_url = 'http://onlinepaymentprocessing.com/'
+      self.display_name = 'AGMS Gateway'
+
+      STANDARD_ERROR_CODE_MAPPING = {
+        "2" => STANDARD_ERROR_CODE[:card_declined],
+        "10" => STANDARD_ERROR_CODE[:processing_error],
+        "20" => STANDARD_ERROR_CODE[:processing_error],
+        "30" => STANDARD_ERROR_CODE[:processing_error],
+      }
+
+      def initialize(options={})
+        requires!(options, :login, :password)
```

Comment:
Credential names should map as closely as possible to the gateway's nomenclature, so I'd use `gateway_username` and `gateway_password`.